Shared/Samples/Authenticate with token/AuthenticateWithTokenView.swift

```diff
@@ -0,0 +1,132 @@
+// limitations under the License.
+
+import ArcGIS
+import ArcGISToolkit
+import SwiftUI
+
+struct AuthenticateWithTokenView: View {
+    /// The authenticator to handle authentication challenges.
+    @StateObject private var authenticator = Authenticator()
+    
+    /// The loaded map result.
+    @State private var mapLoadResult: Result<Map, Error>?
+    
+    /// Creates the map for this sample.
+    static func makeMap() -> Map {
+        // The portal to authenticate with named user.
+        let portal = Portal(url: .portal, connection: .authenticated)
+        
+        // The portal item to be displayed on the map.
+        let portalItem = PortalItem(
+            portal: portal,
+            id: .trafficMap
+        )
+        
+        // Creates map with portal item.
+        return Map(item: portalItem)
+    }
+    
+    var body: some View {
+        Group {
+            if let mapLoadResult = mapLoadResult {
+                switch mapLoadResult {
+                case .success(let value):
+                    MapView(map: value)
+                case .failure(let error):
+                    Text("Error loading map: \(errorString(for: error))")
+                        .padding()
+                }
+            } else {
+                ProgressView()
+                    .task {
+                        mapLoadResult = await Result { @MainActor in
+                            let map = Self.makeMap()
+                            try await map.load()
+                            try await map.operationalLayers.first?.load()
+                            return map
+                        }
+                    }
+            }
+        }
+        .authenticator(authenticator)
+        .onAppear {
+            // Setting the challenge handlers here in `onAppear` so user is prompted to enter
+            // credentials every time trying the sample. In real world applications, set challenge
+            // handlers at the start of the application.
+            
+            // Sets authenticator as ArcGIS and Network challenge handlers to handle authentication
+            // challenges.
+            ArcGISEnvironment.authenticationManager.handleChallenges(using: authenticator)
+            
+            // In real world applications, uncomment this code to persist credentials in the
+            // keychain and remove `signOut()` from `onDisappear`.
+            // setupPersistentCredentialStorage()
+        }
+        .onDisappear {
+            // Resetting the challenge handlers and clearing credentials here in `onDisappear`
+            // so user is prompted to enter credentials every time trying the sample. In real
+            // world applications, do these from sign out functionality of the application.
+            
+            // Resets challenge handlers.
+            ArcGISEnvironment.authenticationManager.handleChallenges(using: nil)
+            
+            signOut()
```

Comment:
Please follow IWA and PKI samples and keep the same code in separate functions. There is some discussion in PKI sample for keeping the revoke tokens function as well.